cmd/vl3-nse/service_registry.go

```diff
@@ -23,12 +26,30 @@ const (
 
 type validationErrors []error
 
-func NewServiceRegistry(addr string) (ServiceRegistry, ServiceRegistryClient, error) {
-	conn, err := grpc.Dial(addr,
-		grpc.WithInsecure(),
+func NewServiceRegistry(addr string, ctx context.Context) (ServiceRegistry, ServiceRegistryClient, error) {
+	opts := []grpc.DialOption{
 		grpc.WithUnaryInterceptor(grpc_prometheus.UnaryClientInterceptor),
 		grpc.WithStreamInterceptor(grpc_prometheus.StreamClientInterceptor),
-	)
+	}
+
+	insecure, err := strconv.ParseBool(os.Getenv(tools.InsecureEnv))
+	if err != nil {
+		logrus.Error("Missing INSECURE env variable.")
+		logrus.Info("Continuing with insecure mode.")
+		insecure = true
+	}
+
+	if !insecure && tools.GetConfig().SecurityProvider != nil {
+		if tlsConfig, err := tools.GetConfig().SecurityProvider.GetTLSConfig(ctx); err != nil {
+			opts = append(opts, grpc.WithInsecure())
```

Comment:
Please add an error log here to indicate there's a problem with the tlsconfig and the error code.